### PR TITLE
Add documentation

### PR DESCRIPTION
**Pull request by emilyalbini**
@Mark-Simulacrum let me know if there are things you'd like to see covered in more details.

[Rendered](https://github.com/rust-lang/crates-io-heroku-metrics/blob/docs/README.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,142 @@
...
+
+## Design
+
+The container uses [nginx] to route and protect requests, and [Vector] to
+ingest log messages and extract metrics out of them. Two password-protected
+HTTP endpoints are exposed:
+
+* `/drain`, which receives real-time log messages from [Heroku Logplex].
+* `/metrics`, which exposes the processed Prometheus metrics ready to be
+  scraped.
```
Comment (Mark-Simulacrum) [inline README.md:99]: I think it'd be good to document the kind of request rate / style we're expecting here -- is this one HTTP request per each message emitted? I don't think the rate itself is too important but that'll help with a high-level understanding I think.

File: README.md [hunk 2]
```diff
@@ -1,3 +1,142 @@
...
+
+How metrics are extracted depends on the kind of metric.
+
+### Instance-level metrics
+
+To gather instance-level metrics the container expects log messages in a
+specific format. Each line must be prefixed with the
+`crates-io-heroku-metrics:ingest` prefix, and must then contain the
+Base64-encoded JSON-encoded metrics data. To simplify the logic inside the
```
Comment (Mark-Simulacrum) [inline README.md]: Can you say more about why we need to base-64 encode the JSON? That seems a bit surprising and will increase log volumes...

**Review by Mark-Simulacrum — COMMENTED** [review]
Looks good to me - left a few comments.

**Comment by emilyalbini** [comment]
Addressed review feedback.

**Review by Mark-Simulacrum — APPROVED** [review]

**Pull request merged**